client/homepage/layout.js

```diff
@@ -34,43 +56,113 @@ const Layout = () => {
 		};
 	}, [] );
 
+	const { query, requestingTaskList, taskListHidden } = props;
+	const isTaskListEnabled = taskListHidden === false;
+	const isDashboardShown = ! isTaskListEnabled || ! query.task;
+
+	const renderColumns = () => {
+		return (
+			<Fragment>
+				{ showInbox && (
+					<div className="woocommerce-homepage-column is-inbox">
+						<div className="temp-content">
+							<Button
+								isPrimary
+								onClick={ () => {
+									setShowInbox( false );
+								} }
+							>
+								Dismiss All
+							</Button>
+						</div>
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+					</div>
+				) }
+				<div
+					className="woocommerce-homepage-column"
+					ref={ content }
+					style={ {
+						position: isContentSticky ? 'sticky' : 'static',
+					} }
+				>
+					{ isTaskListEnabled && renderTaskList() }
+					<StatsOverview />
+				</div>
+			</Fragment>
+		);
+	};
+
+	const renderTaskList = () => {
+		if ( requestingTaskList ) {
+			return <TaskListPlaceholder />;
+		}
+
+		return (
+			<Suspense fallback={ <Spinner /> }>
+				<TaskList
+					query={ query }
+					inline
+				/>
+			</Suspense>
+		);
+	};
+
 	return (
 		<div
 			className={ classnames( 'woocommerce-homepage', {
 				hasInbox: showInbox,
 			} ) }
 		>
-			{ showInbox && (
-				<div className="woocommerce-homepage-column is-inbox">
-					<div className="temp-content">
-						<Button
-							isPrimary
-							onClick={ () => {
-								setShowInbox( false );
-							} }
-						>
-							Dismiss All
-						</Button>
-					</div>
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-				</div>
-			) }
-			<div
-				className="woocommerce-homepage-column"
-				ref={ content }
-				style={ {
-					position: isContentSticky ? 'sticky' : 'static',
-				} }
-			>
-				<StatsOverview />
-			</div>
+			{ isDashboardShown
+				? renderColumns()
+				: isTaskListEnabled && renderTaskList()
+			}
 		</div>
 	);
 };
 
-export default Layout;
+Layout.propTypes = {
+	/**
+	 * If the task list option is being fetched.
+	 */
+	requestingTaskList: PropTypes.bool.isRequired,
+	/**
+	 * If the task list is hidden.
+	 */
+	taskListHidden: PropTypes.bool,
+	/**
+	 * Page query, used to determine the current task if any.
+	 */
+	query: PropTypes.object.isRequired,
+};
+
+export default compose(
+	withSelect( ( select ) => {
+		const {
+			getOptions,
+			isGetOptionsRequesting,
+		} = select( 'wc-api' );
+
+		if ( isOnboardingEnabled() ) {
+			const options = getOptions( [
+				'woocommerce_task_list_hidden',
+			] );
```

Comment:
We can preload this option but that doesn't remove the need for the placeholder - it's in use as the Suspense fallback as well.

Reply:
Sounds good, 👍  to preloading

Reply:
It looks like this is already preloaded: https://github.com/woocommerce/woocommerce-admin/blob/5f3223fa8313b7ccd363ed02f09e2fe5e8b0b83a/src/Features/Onboarding.php#L504

Reply:
Is that file getting loaded on the homescreen route? I think we do conditional loading of OBW logic. I haven't fired this one up, but I remember it waiting on this option to be fetched.

Reply:
Nevermind, that option is available at `wcSettings.preloadOptions.woocommerce_task_list_hidden`.

I'll take a closer look today.

Reply:
Its some kind of wc-api thing because if I hard code the values, then it renders as soon as the chunk has arrived. This looks fine to me, #4144 will ensure the preloaded option is available immediately.